rewrite-java/src/main/java/org/openrewrite/java/VariableNameUtils.java

```diff
@@ -243,6 +243,17 @@ public J.ClassDeclaration visitClassDeclaration(J.ClassDeclaration classDecl, Se
             return super.visitClassDeclaration(classDecl, namesInScope);
         }
 
+        @Override
+        public J.InstanceOf visitInstanceOf(J.InstanceOf instanceOf, Set<String> strings) {
+            if (instanceOf.getPattern() instanceof J.Identifier) {
+                Set<String> names = nameScopes.get(currentScope.peek());
+                if (names != null) {
+                    names.add(((J.Identifier)instanceOf.getPattern()).getSimpleName());
+                }
+            }
+            return super.visitInstanceOf(instanceOf, strings);
```

Comment:
@knutwannheden is this what you meant by adding it to the parent scope?
```suggestion
        public J.InstanceOf visitInstanceOf(J.InstanceOf instanceOf, Set<String> namesInScope) {
            if (instanceOf.getPattern() instanceof J.Identifier) {
                Set<String> names = nameScopes.get(currentScope.peek());
                if (names != null) {
                    String simpleName = ((J.Identifier) instanceOf.getPattern()).getSimpleName();
                    names.add(simpleName);
                    namesInScope.add(simpleName);
                }
            }
            return super.visitInstanceOf(instanceOf, namesInScope);
```

Reply:
Not exactly sure what the effect would be of this change. But For example the following code must not end up creating two pattern variables with the same name, as that would give a compile error:
```java
        Object o = "";
        if (!(o instanceof String)) {
            throw new IllegalArgumentException(((String) o));
        }
        if (o instanceof String) {
            System.out.println((String) o);
        }
```

Reply:
The given example is very contrived, as it would result in a runtime exception. But I wanted to get the point across regarding the compiler error. I could otherwise also come up with more sensible examples.

So, to be on the safe side, a pattern variable as in the following code should be regarded as belonging to the surrounding block (not just the if statement's then block):
```java
        Object o = "";
        if (o instanceof String s) {
            // ...
        }
```

That way the `VariableNameUtils` should not propose `s` as a variable name in any statements following the `if` statement.

Reply:
Looking this over again and how it's used in InstanceOfPattern match I _think_ right now we don't need the above suggested fix, given how we aggregate the scope names in postVisit
https://github.com/openrewrite/rewrite/blob/0f0dc80f793d0e33f60e1e982318d82c40cc4201/rewrite-java/src/main/java/org/openrewrite/java/VariableNameUtils.java#L193-L213


I'll try to pick this up in rewrite-static-analysis.